src/Models/Message.php

```diff
@@ -52,6 +65,21 @@ public function conversation(): BelongsTo
         return $this->belongsTo(Conversation::class);
     }
 
+    /**
+     * Scope a query to only include messages of the given conversation.
+     *
+     * @param  \Illuminate\Database\Eloquent\Builder  $query
+     * @param  \OwowAgency\Gossip\Models\Conversation  $conversation
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopeOfConversation($query, Conversation $conversation)
+    {
+        return $query->whereHas(
+            'conversation',
+            fn($query) => $query->where('conversations.id', $conversation->id),
+        );
+    }
```

Comment:
Might also apply on your previous PR. Much more efficient to query just the `conversation_id` of this model. 
```php
    public function scopeOfConversation($query, Conversation $conversation)
    {
        return $query->where('conversation_id', $conversation->id);
    }
```

Reply:
I think I was sleeping 😴 .. I also removed the type hint because the model can change in the configuration value.